vagrant/lib/forklift/box_distributor.rb

```diff
@@ -60,13 +60,31 @@ def distribute!
       end
     end
 
+    def fetch_from_box_or_settings(box, key, default_value)
+      if !(value_from_box = box.fetch(key, nil)).nil?
+        value_from_box
+      else
+        @settings.fetch(key, default_value)
+      end
+    end
+
+    def ssh_config_keys_default_values
+      [%w[username vagrant], ['forward_agent', false], ['keys_only', true]]
```

Comment:
Feels like this could be a a hash
```suggestion
      {
        username: 'vagrant',
        forward_agent: false,
        keys_only: true,
      }
```

Also, I'd consider not setting the SSH username and relying on Vagrant to take care of that.

Reply:
I tried this initially but couldn't find a satisfactory solution. Looking again, I think https://apidock.com/ruby/v2_6_3/Hash/each_pair is what I need

Really, Vagrant should be able to set defaults for all of these. I initially tried to copy what looked like the intent of the existing pattern with `forward_agent` but looking again at https://www.vagrantup.com/docs/vagrantfile/ssh_settings I don't see any reason why we'd need to have defaults here.

I will try a version without them.